defs/go.bzl

```diff
@@ -27,61 +25,34 @@ def _compute_genrule_variables(resolved_srcs, resolved_outs):
         variables["@"] = list(resolved_outs)[0].path
     return variables
 
+# Creates a command sequence which sets Go environment variables
```

Comment:
Since this has been reduced to setting environment variables, you should be able to use the `env=` param of `ctx.action()` instead of constructing a shell blob. You'll want to adjust the existing `env=ctx.configuration.default_shell_env,` line to use a dict containing the Go variables.

Reply:
Thanks for the feedback!

I tried going with this all the way, but the paths provided by bazel are relative paths, which go doesn't like, and I don't think there's any way to get the absolute path to the execdir from skylark (perhaps by design).

This means that some of the environment variables can be passed through `env=` (and I've updated this PR to use that), but `GOPATH` and `GOROOT` must still be set in a shell blob to get access to `$(pwd)`.